contentctl/objects/savedsearches_conf.py

```diff
@@ -0,0 +1,185 @@
+        :param line: a line from the conf file
+        """
+        # Determine the new section name:
+        new_section_name = line.strip().strip("[").strip("]")
+
+        # Raise if we are in a section already according to the state (we cannot statr a new section
+        # before ending the previous section)
+        if self._in_section:
+            raise Exception(
+                "Attempting to start a new section w/o ending the current one; check for "
+                f"parsing/serialization errors: (current section: '{self._current_section_name}', "
+                f"new section: '{new_section_name}') [see line {self._current_line_no} in "
+                f"{self.path}]"
+            )
+
+        # Capture the name of this section, reset the lines, and indicate that we are now in a
+        # section
+        self._current_section_name = new_section_name
+        self._current_section_lines = [line]
+        self._in_section = True
+
+    def section_end(self) -> None:
+        """
+        Adjust the state end the section we were enumerating; parse the lines as a DetectionStanza
+        """
+        # Name should have been set during section start
+        if self._current_section_name is None:
+            raise Exception(
+                "Name for the current section was never set; check for parsing/serialization "
+                f"errors [see line {self._current_line_no} in {self.path}]."
+            )
+        elif self._current_section_name in self.detection_stanzas:
+            # Each stanza should be unique, so the name should not already be in the dict
+            raise Exception(
+                f"Name '{self._current_section_name}' already in set of stanzas [see line "
+                f"{self._current_line_no} in {self.path}]."
+            )
+
+        # Build the stanza model from the accumulated lines and adjust the state to end this section
+        self.detection_stanzas[self._current_section_name] = DetectionStanza(
+            name=self._current_section_name,
+            lines=self._current_section_lines
+        )
+        self._in_section = False
+
+    def _parse_detection_stanzas(self) -> None:
+        """
+        Open the conf file, and parse out DetectionStanza objects from the raw conf stanzas
+        """
+        # We don't want to parse the stanzas twice (non-atomic operation)
+        if len(self.detection_stanzas) != 0:
+            raise Exception(
+                f"{len(self.detection_stanzas)} stanzas have already been parsed from this conf; we"
+                " do not need to parse them again"
+            )
+
+        # Open the conf file and iterate over the lines
+        with open(self.path, "r") as file:
+            for line in file:
+                self._current_line_no += 1
+
+                # Break when we get to the end of the ESCU detections
+                if line.strip() == "### END ESCU DETECTIONS ###":
```

Comment:
If we want this to work on non escu content, then it needs to be derived from the APP NAME:
https://github.com/splunk/contentctl/blob/b9ce7f9e75b0a8bfbc0ab13274a6be01e2642be0/contentctl/output/templates/savedsearches_detections.j2#L114

This can be marked as an enhancement if you agree since for some time we will be the only users of this feature in which case ESCU is appropriate

Reply:
Resolving this as a future TODO with other changes that support non-escu apps from taking advantage of versioning: https://github.com/splunk/contentctl/issues/283

Reply:
Unresolving based on my comment here: https://github.com/splunk/contentctl/pull/280#discussion_r1752947841

Reply:
The label for the app already exists in the detection parser:
https://github.com/splunk/contentctl/pull/280/files#diff-07381b57e93d8700a66523fb3b3258e6dcb3aa4e19d2feceab8e0f50076feed6R24
Let's just use that for the break like
`if line.strip() == f"### END {self.app_label} DETECTIONS ###":`
instead of
`if line.strip() == "### END ESCU DETECTIONS ###":`

You can see where that name comes from 
https://github.com/splunk/contentctl/blob/b9ce7f9e75b0a8bfbc0ab13274a6be01e2642be0/contentctl/output/templates/savedsearches_detections.j2#L114 
and 
https://github.com/splunk/contentctl/blob/b9ce7f9e75b0a8bfbc0ab13274a6be01e2642be0/contentctl/output/conf_writer.py#L142

I will leave this one open until that change is made, if you agree,

Reply:
@pyth0n1c changes made! Please resolve if they look good

Reply:
This looks great, thanks for the change!